source/rafcon/gui/helpers/coordinate_translation.py

```diff
@@ -0,0 +1,124 @@
+    """
+    The CoordinateTranslator is a helper class, helping to transform coordinates between different coordinate systems
+    (e.g screen, item, window, and view coordinates) in the GUI.
+    """
+    main_window = None
+    sm_notebook_page = None
+    graphical_editor_controller = None
+
+    def __init__(self):
+        self.main_window = main_window_controller.view.get_top_widget()
+        sm_controllers = main_window_controller.state_machines_editor_ctrl
+        self.sm_notebook_page = list(sm_controllers.tabs.items())[0][1]['page']
+        self.graphical_editor_controller = sm_controllers.get_controller(1)
+
+    def main_window_coordinates_to_notebook_coordinates(self, main_window_coordinates):
+        """
+        Transforms a point relative to the main window, into a point relative to the state machine notebook.
+        (window coordinates --> view coordinates).
+        :param (float, float) main_window_coordinates:
+        A tuple containing x and y coordinates relative to the main window.
+        :return: The same point, but relative to the state machine notebook (That's the view coordinate system).
+        :rtype: (float, float)
+        """
+        return self.main_window.translate_coordinates(self.sm_notebook_page, main_window_coordinates[0],
+                                                      main_window_coordinates[1])
+
+    def notebook_coordinates_to_main_window_coordinates(self, notebook_coordinates):
+        """
+        Transforms a point relative to the state machine notebook, into a point relative to the main window.
+        (view coordinates --> window coordinates)
+        :param (float, float) notebook_coordinates:
+        A tuple containing a x and a y coordinate relative to the state machine notebook.
+        :return: The same point, but relative to the main window.
+        :rtype: (float, float)
+        """
+        return self.sm_notebook_page.translate_coordinates(self.main_window, notebook_coordinates[0],
+                                                           notebook_coordinates[1])
+
+    def notebook_coordinates_to_item_coordinates(self, state_machine_m, target_state_m, notebook_coordinates):
+        """
+        Transforms a point relative to the state machine notebook, into a point relative to the target state model.
+        (view coordinates --> item coordinates)
+        :param StateMachineModel state_machine_m: The state machine model, the target state model belongs to.
+        :param StateModel target_state_m: The state model, the resulting point should be relative to.
+        :param (float,float) notebook_coordinates:
+        A tuple containing a x and a y coordinate relative to the state machine notebook.
+        :return: The same point, relative to the target state model.
+        :rtype: (float,float)
+        """
+        root_state = self.__get_root_state(target_state_m.state)
+        root_state_m = state_machine_m.get_state_model_by_path(root_state.get_path())
```

Comment:
why not implement `get_root_state` to directly take a `StateModel`, then you would not need to convert from model to core and back to model.